### DISCUSSION TITLE
Can I configure i18n fallback on my own?

### DISCUSSION THREAD

**Discussion by nnecec** [withastro/starlight#1354] [Feature Requests]
### What version of `starlight` are you using?

0.15.2

### What is your idea?

Make i18n fallback as a configuration, not fixed to `true`

### Why is this feature necessary?

When I'm writing i18n docs, there are difference between two language files. I just don't need these fallback pages on the second language, but they are showed at sidebar.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the feature request @nnecec! Could you describe your use case in more detail please?

What are the differences between the different languages? Why are they different? How would you like the sidebar to behave in each language? What should happen when using the language switcher to switch between two languages when a page doesn’t exist in both?

**Reply by nnecec** [reply]
1. Why ?
	I processed [the book](https://github.com/gibbok/typescript-book) into multiple md files and [deployed them using astro](https://typescript-book-git-dev-nnecec.vercel.app/zh-cn/books/2-translations/) recently. The English version and the Chinese version were not synchronized in real time, resulting in a different number of chapters that could not correspond one to one. Therefore, I cannot match English file names in order for Chinese md files in the processing code. **Of course, this is an unusual situation. If you think this situation should not be handled, you can ignore my question :)**

2. How would you like the sidebar to behave in each language?
	Fallback should be a default behavior, it's cool and useful in most cases. But it would be nice to give me a choice whether to turn it on or not. If I turn fallback off, starlight shows current language files for selected language.
	
3. What happened when switch between two languages ?
	I'm trying to switch languages right now, it'll redirect to 404 page. I think it's OK, because starlight and me both don't know `/en/apple` corresponds to `/zh-cn/苹果`. If developer want deal the situation, he can redirect the 404 page to the home page.

**Reply by delucis** [reply]
> The English version and the Chinese version were not synchronized in real time, resulting in a different number of chapters that could not correspond one to one.

Thanks for explaining — makes total sense!

> If I turn fallback off, starlight shows current language files for selected language.

I assume here you by “shows current language files” you mean in the sidebar. The tricky part here is we don’t have a way to configure sidebars differently for each language. While in theory if using a sidebar that is entirely auto-generated, it could be possible to do this, we would also need to support explicit sidebar configuration, which could get pretty complex if every language has its own slight variations.

> it'll redirect to 404 page. I think it's OK

👍 Good to know.

---

It seems like it would be slightly tricky to support this use-case out of the box, but you may be able to handle this today by [overriding components](https://starlight.astro.build/guides/overriding-components/) and mutating the [`sidebar`](https://starlight.astro.build/reference/overrides/#sidebar) and [`pagination`](https://starlight.astro.build/reference/overrides/#pagination) props to avoid the fallback pages showing up.
